vendor/bat-native-ads/src/bat/ads/internal/tab_manager/tab_manager_unittest.cc

```diff
@@ -0,0 +1,356 @@
+}
+
+TEST_F(BatAdsTabManagerTest,
+    BrowserWindowDidEnterBackgroundUserActivityEvent) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnBackgrounded();
+
+  // Assert
+  const int count = UserActivityCountForEventType(
+      UserActivityEventType::kBrowserWindowDidEnterBackground);
+
+  EXPECT_EQ(1, count);
+}
+
+TEST_F(BatAdsTabManagerTest,
+    IsTabVisible) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com", true, false);
+
+  // Assert
+  EXPECT_TRUE(TabManager::Get()->IsVisible(1));
+}
+
+TEST_F(BatAdsTabManagerTest,
+    IsTabOccluded) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com", false, false);
+
+  // Assert
+  EXPECT_FALSE(TabManager::Get()->IsVisible(1));
+}
+
+TEST_F(BatAdsTabManagerTest,
+    UpdatedTab) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com", true, false);
+
+  // Assert
+  base::Optional<TabInfo> tab = TabManager::Get()->GetForId(1);
+
+  TabInfo expected_tab;
+  expected_tab.id = 1;
+  expected_tab.url = "https://brave.com";
+  expected_tab.is_playing_media = false;
+  EXPECT_EQ(expected_tab, tab);
+}
+
+TEST_F(BatAdsTabManagerTest,
+    UpdatedTabUserActivityEvent) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com", true, false);
+
+  // Assert
+  const int count = UserActivityCountForEventType(
+      UserActivityEventType::kOpenedNewOrFocusedOnExistingTab);
+
+  EXPECT_EQ(1, count);
+}
+
+TEST_F(BatAdsTabManagerTest,
+    UpdatedIncognitoTab) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com", true, true);
+
+  // Assert
+  EXPECT_EQ(0, UserActivityCount());
+
+  base::Optional<TabInfo> tab = TabManager::Get()->GetForId(1);
+  EXPECT_EQ(base::nullopt, tab);
+}
+
+TEST_F(BatAdsTabManagerTest,
+    UpdatedOccludedTab) {
+  // Arrange
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com", false, false);
+
+  // Assert
+  EXPECT_EQ(0, UserActivityCount());
+
+  base::Optional<TabInfo> tab = TabManager::Get()->GetForId(1);
+  EXPECT_EQ(base::nullopt, tab);
+}
+
+TEST_F(BatAdsTabManagerTest,
+    UpdatedExistingTab) {
+  // Arrange
+  TabManager::Get()->OnUpdated(1, "https://brave.com", true, false);
+
+  // Act
+  TabManager::Get()->OnUpdated(1, "https://brave.com/about", true, false);
+
+  // Assert
+  EXPECT_EQ(1, UserActivityCount());
+
+  base::Optional<TabInfo> tab = TabManager::Get()->GetForId(1);
+
+  TabInfo expected_tab;
+  expected_tab.id = 1;
+  expected_tab.url = "https://brave.com/about";
+  expected_tab.is_playing_media = false;
+  EXPECT_EQ(expected_tab, tab);
+}
+
+TEST_F(BatAdsTabManagerTest,
+    ClosedTab) {
+  // Arrange
+  TabManager::Get()->OnUpdated(1, "https://brave.com", true, false);
+
+  // Act
+  TabManager::Get()->OnClosed(1);
+
+  // Assert
+  EXPECT_EQ(2, UserActivityCount());
+
+  base::Optional<TabInfo> tab = TabManager::Get()->GetForId(1);
+  EXPECT_EQ(base::nullopt, tab);
```

Comment:
@tmancey just double checking how we are verifying that the tabs manager not indefinitely filling up the `tabs_` map (in case tabs would never get deleted). But I think that does it.

Reply:
This covers this case, thanks